src/observers/infoMessageObserver.ts

```diff
@@ -71,4 +83,62 @@ export class InfoMessageObserver {
                 ));
         }
     }
+
+    private handleExtensionActivated() {

```

Comment:
Can we try to extract the duplicate code with the function `handleCommandWithUnkownUserTypeTriggered` to a common function with parameters to handle the different part.

Reply:
Thought about it. but the parameter will be a super long string, or do you have any good suggestion?

Reply:
does that make any difference? for now, you just call the `showInfoMessage` with a long string, and as you can see, there is an almost the same code segment inside the two function.
how about extract a function `handleUserTypeSelection`:
```
function handleUserTypeSelection(String msg) {
   this.showInfoMessage(
                msg +
                `You are still able to change it by the extension settings (Docs validation -> User type) after this selection`,
                new MessageAction(
                    "Microsoft internal employee",
                    undefined,
                    undefined,
                    () => {
                        const extensionConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(EXTENSION_NAME, undefined);
                        extensionConfig.update(USER_TYPE, UserType.MicrosoftInternalEmployee, true);
                    }
                ),
                new MessageAction(
                    "Public contributor",
                    undefined,
                    undefined,
                    () => {
                        const extensionConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(EXTENSION_NAME, undefined);
                        extensionConfig.update(USER_TYPE, UserType.PublicContributor, true);
                    }
                ));
}

```